Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -108,6 +131,48 @@ public void Serialize(IIonWriter writer, object item)
             writer.StepOut();
         }
 
+        private object InvokeIonConstructor(ConstructorInfo ionConstructor, IIonReader reader)
+        {
+            var parameters = ionConstructor.GetParameters();
+            var arguments = new object[parameters.Length];
+
+            if (parameters.Length > 0)
+            {
+                // Compute mapping between parameter names and index in parameter array.
+                var paramIndexMap = new Dictionary<string, int>();
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    var param = (IonPropertyName)parameters[i].GetCustomAttribute(typeof(IonPropertyName));
+                    if (param == null)
+                    {
+                        throw new NotSupportedException(
+                            $"Parameter '{parameters[i].Name}' is not specified with the [IonPropertyName] attribute " +
```

Comment:
Append "all constructor arguments must be annotated so that we know which parameters to set at construction time"

Reply:
Done.